src/documentCreator/contexts/FormDataContext/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement, ReactNode } from "react";
+import { issueDocument } from "@govtechsg/tradetrust-schema";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { Document } from "@govtechsg/decentralized-renderer-react-components";
+
+interface FormDataState {
+  documentsList: Array<Document>;
+  setDocumentsList: (documentsList: Document[]) => void;
+  wrappedDocument: WrappedDocument<any> | null;
+  setDocument: (document: Document) => void;
+}
+
+export const FormDataContext = React.createContext<FormDataState>({
+  documentsList: [],
+  setDocumentsList: (documentsList: Document[]) => documentsList,
+  wrappedDocument: null,
+  setDocument: (document: Document) => document
+});
+
+interface FormDataProps {
+  children: ReactNode;
+}
+
+export class FormDataProvider extends React.Component<FormDataProps, FormDataState> {
+  constructor(props: FormDataProps) {
```

Comment:
Wondering why you would not prefer to use hooks instead. You can manage each individual state separately, while setting the default within the the hook itself.